src/bioregistry/data/bioregistry.json

```
@@ -11931,7 +11931,31 @@
       "is_obo": false,
       "prefix": "DOI",
       "uri_format": "http://dx.doi.org/$1"
-    }
+    },
+    "providers": [
+      {
+        "code": "dx_doi_https",
+        "description": "An alternate provider from the DOI website using HTTPS",
+        "homepage": "https://www.doi.org",
+        "name": "Digital Object Identifier",
+        "uri_format": "https://dx.doi.org/$1"
+      },
+      {
+        "code": "doi_https",
+        "description": "An alternate provider from the DOI website",
+        "homepage": "https://www.doi.org",
+        "name": "Digital Object Identifier",
+        "uri_format": "https://doi.org/$1"
+      },
+      {
+        "code": "doi_http",
+        "description": "An alternate provider from the DOI website",
+        "homepage": "https://www.doi.org",
+        "name": "Digital Object Identifier",
+        "uri_format": "http://doi.org/$1"
+      }
+    ],
+    "uri_format": "http://dx.doi.org/$1"
```

I like the previous uri format for DOIs:

```suggestion
    "uri_format": "https://doi.org/$1"
```

Since that is supposedly "preferred" at least for humans, but I guess not for linked data as per https://github.com/biopragmatics/bioregistry/issues/287.

---

> In https://github.com/biopragmatics/bioregistry/pull/291, I'm trying to make the alternate resolvers more accessible for generating URIs

Sounds like a good idea, although the challenge seems to be that there are two uses for the URI:

1. as a web address that users can go to. These should do things like use HTTPS when possible and be human readable if possible.
2. as a URI identifier for linked data where following the standard is most important even if that results in a legacy (or non-operational) URL.

It's impractical for the user to know which provider achieves which objective for every prefix. So what would be ideal is if the user could do something like: `bioregistry.get_iri("prefix:id", usage="linked_data")` or `bioregistry.get_iri("prefix:id", usage="webpage")`. This would require major work, not saying it's worth it.